### PR TITLE
CodeFixProvider for CA2242

### LINKED ISSUE #553: Port FxCop rule CA2242: TestForNaNCorrectly
**Title:** Test for NaN correctly

**Description:**

This expression tests a value against Single.Nan or Double.Nan. Use Single.IsNan(Single) or Double.IsNan(Double) to test the value.

**Dependency:** IOperation

**Notes:**

### PR DESCRIPTION
**Pull request by bkoelman**
Fixes #553.

### CODE REVIEW

**Comment by bkoelman** [comment]
Tagging @lgolding @dpoeschl @srivatsn @JohnHamby for review.

**Comment by bkoelman** [comment]
I have resolved merge conflicts. Is anything needed from me to get this merged?

File: src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs [hunk 1]
```diff
@@ -5,11 +5,540 @@
...
+Public Class A
+    Public Function Compare(s As Single) As Boolean
+        Return Single.NaN = s
+    End Function
+End Class
+", @"
+Public Class A
+    Public Function Compare(s As Single) As Boolean
+        Return Single.IsNaN(s
+)
```
Comment (mavasani) [inline src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs]: Why is the closing bracket on a new line?

File: src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs [hunk 2]
```diff
@@ -5,11 +5,540 @@
...
+Public Class A
+    Public Function Compare() As Boolean
+        Return Double.NaN = Double.NaN
+    End Function
+End Class
+", @"
+Public Class A
+    Public Function Compare() As Boolean
+        Return Double.IsNaN(Double.NaN
+)
```
Comment (mavasani) [inline src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs]: Same here.

**Comment by mavasani** [comment]
👍

File: src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs (hunk 1 above)
Comment (bkoelman) [inline src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs]: I wish it were different, but that is what the formatter does today...

Comment (mavasani) [inline src/System.Runtime.Analyzers/UnitTests/TestForNaNCorrectlyTests.Fixer.cs]: Sure no issues

**Pull request merged**